sound/soc/sof/ipc4-topology.c

```diff
@@ -1066,6 +1066,10 @@ static int sof_ipc4_init_audio_fmt(struct snd_sof_dev *sdev,
 		return -EINVAL;
 	}
 
+	/* pick the only available input format */
```

Comment:
But what if the only available input format doesn't match the input format? Shouldn't we return error?

Reply:
No, I dont think so. The topology would restrict the format to 1 only so that it can be chosen unconditionally. If it is incorrect, you'd have distortion.

Reply:
> No, I dont think so. The topology would restrict the format to 1 only so that it can be chosen unconditionally. If it is incorrect, you'd have distortion.

For output format, yes, we can choose it unconditionally because the widget can do some conversion inside. But for input format, doesn't the only one available input format mean we should always provide the format as the input format? Who will do the conversion if the input format doesn't match the available format? 
For example, the output format of previous widget is s16_le/48000Hz/2ch, but the only available input format of this widget is s32_le/48000Hz/2ch and we set it as the input format of this widget. I don't think firmware will accept it.
We at least need a warning to tell people the input format is not as expected.

Reply:
@bardliao warning is a bit too much. Think about the case of capture. If we capture 16-bit but the topology defines the only supported input format for the DAI copier as 32-bit, you'll see the warning here even though tthere's no issues because the host copier would handle the conversion.

Reply:
> @bardliao warning is a bit too much. Think about the case of capture. If we capture 16-bit but the topology defines the only supported input format for the DAI copier as 32-bit, you'll see the warning here even though tthere's no issues because the host copier would handle the conversion.

Hmm, you convinced me.